src/account/account.service.ts

```diff
@@ -153,9 +153,15 @@ export class AccountService {
 
         // We can safely assume that there is an account for the user due to
         // the foreign key constraint on the users table
-        return await this.db(TABLE_ACCOUNTS)
+        const account = await this.db(TABLE_ACCOUNTS)
             .where('id', user.account_id)
             .first();
+
+        if (!account) {
```

Comment:
ooh nice, i didn't think about putting this here, but it actually makes a ton of sense seeming though its unexpected that a site doesn't have a default account

Reply:
I think there might be a couple of places where we do this check outside of the service, could we clean those up as part of this PR?

Reply:
also could we add a test case for this change in functionality?